------------------------------------------------------------------------
r626814 | mueller | 2007-01-24 18:16:52 +0100 (Wed, 24 Jan 2007) | 5 lines
Changed paths:
   M /branches/KDE/3.5/kdeaddons/konq-plugins/webarchiver/archivedialog.cpp
   M /branches/KDE/3.5/kdeaddons/konq-plugins/webarchiver/webarchivecreator.cpp

fix a minor but annoying privacy issue: the web archiver
would happily put local files or other launch other unsafe
protocols if referenced in the website, even though konqueror
itself wouldn't handle it

------------------------------------------------------------------------
Index: konq-plugins/webarchiver/webarchivecreator.cpp
===================================================================
--- konq-plugins/webarchiver/webarchivecreator.cpp	(revision 626813)
+++ konq-plugins/webarchiver/webarchivecreator.cpp	(revision 626814)
@@ -32,7 +32,7 @@
 
 extern "C"
 {
-	ThumbCreator *new_creator()
+	ThumbCreator * KDE_EXPORT new_creator()
 	{
 		return new WebArchiveCreator;
 	}
Index: konq-plugins/webarchiver/archivedialog.cpp
===================================================================
--- konq-plugins/webarchiver/archivedialog.cpp	(revision 626813)
+++ konq-plugins/webarchiver/archivedialog.cpp	(revision 626814)
@@ -395,12 +395,15 @@ void ArchiveDialog::saveArchiveRecursive
 QString ArchiveDialog::handleLink(const KURL& _url, const QString& _link)
 {
    KURL url(getAbsoluteURL(_url, _link));
-   QString tarFileName;
 
-   if (m_state==Retrieving)
-      m_urlsToDownload.append(url);
-   else if (m_state==Saving)
-      tarFileName = m_downloadedURLDict[url.url()];
+   QString tarFileName;
+   if (kapp->authorizeURLAction("redirect", _url, url))
+   {
+       if (m_state==Retrieving)
+           m_urlsToDownload.append(url);
+       else if (m_state==Saving)
+           tarFileName = m_downloadedURLDict[url.url()];
+   }
 
    return tarFileName;
 }
